go/mysql/conn.go

```diff
@@ -918,6 +924,23 @@ func (c *Conn) handleNextCommand(handler Handler) bool {
 	return true
 }
 
+func (c *Conn) handleComBinlogDumpGTID(handler Handler, data []byte) (kontinue bool) {
+	defer c.recycleReadPacket()
+
+	c.startWriterBuffering()
+	defer func() {
+		if err := c.endWriterBuffering(); err != nil {
+			log.Errorf("conn %v: flush() failed: %v", c.ID(), err)
+			kontinue = false
+		}
+	}()
+
+	_, _, position, _ := c.parseComBinlogDumpGTID(data)
+	handler.ComBinlogDumpGTID(c, position.GTIDSet)
```

Comment:
if `c.parseComBinlogDumpGTID` fails shouldn't we not continue? or is it already handled by `endWriterBuffering`?

Reply:
good catch!